services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs_getters.py

```diff
@@ -319,7 +319,10 @@ async def get_job_wallet(
     _logger.debug("Getting wallet for job '%s'", job_name)
 
     try:
-        return await webserver_api.get_project_wallet(project_id=job_id)
+        project_wallet = await webserver_api.get_project_wallet(project_id=job_id)
+        if project_wallet:
+            return await webserver_api.get_wallet(wallet_id=project_wallet.wallet_id)
```

Comment:
The wallet id is globally unique, right? So this should guarantee that we are actually getting the correct wallet the second time.

Reply:
correct, the wallet is globally unique :+1: already bound to the product